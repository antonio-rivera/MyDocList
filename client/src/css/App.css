body {
    box-sizing: border-box;
    background-color: rgba(0, 35, 124, 0.75);
}