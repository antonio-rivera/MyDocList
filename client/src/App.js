import React, {useState} from 'react';
import Home from './components/Home'
import AddProvider from './components/AddProvider'
import PatientHome from './components/PatientHome'
import ProviderList from './components/ProviderList'
import AddAppointment from './components/AddAppointment'
import {BrowserRouter as Router, Route, Switch} from 'react-router-dom'
import './css/App.css'

function App() {


  const [message, setMessage] = useState(undefined) //Message that indicates whether the POST request on
                                                    //the AddProvider component was successful or not



  function removeMessage() //Function to make sure after message is delivered it will reset to wait for next request response
  {
    setMessage(undefined)
  }


  function handlePOST_response(res) { //Change message depending on response. Passed as prop to addProvider component

      if(res.status === 201)
        {
          setMessage("Provider added successfully!")
        }

      else 
        {
          setMessage("Some error occurred while adding the provider")
        }

  }

  return (
        <>
        <Router>
        <Switch>
        <Route path='/addAppointment' exact component={AddAppointment}></Route>
            <Route path='/providerList' render={()  => <ProviderList/>}></Route>
            <Route path='/patientHome' render={()  => <PatientHome/>}></Route>
            <Route path='/addProvider' render={() => <AddProvider handlePOST_response={handlePOST_response}/> }></Route>
            <Route path='/' render={() => <Home message={message} removeMessage={removeMessage} /> }></Route>
        </Switch>
        </Router>
        </>
  );
}

export default App;
