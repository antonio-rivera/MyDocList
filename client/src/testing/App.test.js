import React from 'react'
import {shallow} from 'enzyme'
import App from '../App'

describe('Main app component that connects all components', () => {
    it('should render correctly', () => {
        const component = shallow(<App/>)

        expect(component).toMatchSnapshot()
    })
})
