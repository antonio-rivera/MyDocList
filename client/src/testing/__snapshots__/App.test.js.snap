// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main app component that connects all components should render correctly 1`] = `
<Fragment>
  <BrowserRouter>
    <Switch>
      <Route
        path="/appointmentList"
        render={[Function]}
      />
      <Route
        component={[Function]}
        exact={true}
        path="/addAppointment"
      />
      <Route
        path="/providerList"
        render={[Function]}
      />
      <Route
        path="/patientHome"
        render={[Function]}
      />
      <Route
        path="/addProvider"
        render={[Function]}
      />
      <Route
        path="/"
        render={[Function]}
      />
    </Switch>
  </BrowserRouter>
</Fragment>
`;
